src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,144 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
```

Comment:
The `List` interface is not defined in the provided files. Ensure that the `List` interface is implemented correctly and exists in the `core.basesyntax` package.